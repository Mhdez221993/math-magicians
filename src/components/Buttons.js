import React from 'react';

const Buttons = props => {
  const button = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  return (
    <div className="buttons">
      <MapButtons keys={button} cb={props} />
    </div>
  );
};

const MapButtons = props => props.keys.map(key => (
  <div key={key.toString()} className="button">
    <button onClick={props.cb.handleClick} className="btn" type="button" value={key}>{key}</button>
  </div>
));

export default Buttons;
