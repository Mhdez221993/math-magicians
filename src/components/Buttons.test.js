import React from 'react';
import renderer from 'react-test-renderer';
import Buttons from './Buttons';

it('Buttons snapshot test', () => {
  const tree = renderer
    .create(<Buttons />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
