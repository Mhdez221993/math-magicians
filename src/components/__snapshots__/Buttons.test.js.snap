// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons snapshot test 1`] = `
<div
  className="buttons"
>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="AC"
    >
      AC
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="+/-"
    >
      +/-
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="%"
    >
      %
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="รท"
    >
      รท
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="7"
    >
      7
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="8"
    >
      8
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="9"
    >
      9
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="x"
    >
      x
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="4"
    >
      4
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="5"
    >
      5
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="6"
    >
      6
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="-"
    >
      -
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="1"
    >
      1
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="2"
    >
      2
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="3"
    >
      3
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="+"
    >
      +
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="0"
    >
      0
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="."
    >
      .
    </button>
  </div>
  <div
    className="button"
  >
    <button
      className="btn"
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
